File: cpp/Vehicle.cpp
#include "Vehicle.hpp"

// Constructor
Vehicle::Vehicle(const std::string &plate, const std::string &brand, const std::string &clr, int year)
    : number_plate(plate), merk(brand), production_year(year), color(clr) {
}

// Destructor
Vehicle::~Vehicle() {
}

// Getters
std::string Vehicle::getNumberPlate() const {
    return number_plate;
}

std::string Vehicle::getBrand() const {
    return merk;
}

int Vehicle::getProductionYear() const {
    return production_year;
}

std::string Vehicle::getColor() const {
    return color;
}

// Setters
void Vehicle::setNumberPlate(const std::string &plate) {
    number_plate = plate;
}

void Vehicle::setBrand(const std::string &brand) {
    merk = brand;
}

void Vehicle::setProductionYear(int year) {
    production_year = year;
}

void Vehicle::setColor(const std::string &clr) {
    color = clr;
}

File: python/Vehicle.py
class Vehicle:
    def __init__(self, plate, brand, clr, year):
        self._numberPlate = plate
        self._brand = brand
        self._productionYear = year
        self._color = clr

    # Getters
    def getNumberPlate(self):
        return self._numberPlate

    def getBrand(self):
        return self._brand

    def getProductionYear(self):
        return self._productionYear

    def getColor(self):
        return self._color

    # Setters
    def setNumberPlate(self, plate):
        self._numberPlate = plate

    def setBrand(self, brand):
        self._brand = brand

    def setProductionYear(self, year):
        self._productionYear = year

    def setColor(self, clr):
        self._color = clr
